web/concrete/src/Application/Application.php

```diff
@@ -58,9 +58,9 @@ public function shutdown($options = array())
             }
 
             $logger = new Logger();
-            $r = Request::getInstance();
+            $r = $this['request']->getInstance();
```

Comment:
I don't think `$this['request']` is actually set looking at the `HttpServiceProvider`. At this point, we bind `"Concrete\Core\Http\Request"` to the result of `\Concrete\Core\Http\Request::getInstance()`, so we need a couple more changes to actually support `$this['request']` at the moment. I've been thinking about the best pattern to start using to replace these singleton accessors, at the moment I think it's:


```php
<?php

class SomeClassThatUsedToBeASingleton implements LegacySingletonInterface
{

    /** @var callable */
    private static $resolver;

    public static function setSingletonResolver(callable $resolver)
    {
        static::$resolver = $resolver;
    }

    /**
     * @deprecated Instead, use ...
     */
    public static function getInstance()
    {
        $resolver = static::$resolver;
        return $resolver();
    }

}
```

And in its service provider:
```php
<?php

class SomeServiceProvider extends Provider
{

    public function register()
    {
        SomeClassThatUsedToBeASingleton::setSingletonResolver(function() {
            return $this->app->make('SomeClassThatUsedToBeASingleton');
        });
    }

}
```